test/integration/targets/ansible-doc/test.yml

```
@@ -10,6 +10,39 @@
     actual_output_clean: '{{ actual_output.splitlines()[1:] }}'
     expected_output_clean: '{{ expected_output.splitlines()[1:] }}'
   tasks:
+    - name: create temporary file
+      ansible.builtin.tempfile:
+        state: file
+      register: temp_file
+
+    - name: redirect stdout and stderr to file
+      ansible.builtin.shell: |
+        ansible-doc test_docs &> {{ temp_file.path }}
+      args:
+        executable: "{{ ansible_user_shell }}"
```

Is setting `executable` like this really necessary? I don't see any existing use of this pattern in tests, and I'm not sure what it's supposed to accomplish so a comment explaining that would be nice.

If it is necessary, you should use the actual fact name, not rely on variable injection.
```suggestion
        executable: "{{ ansible_facts.user_shell }}"
```

---

Without this variable the integration tests for Ubuntu 24.04 would fail.
```
ansible-test integration --docker ubuntu2404 -v -- ansible-doc
```
For some reason the redirect to a file will not work for this image.\
It will instead land in stdout.\
I could not track it down, Ubuntu should use `bash` by default, but this seems to be some kind of hardened image.\
This was the only way that would work with all containers, `alpine` has no bash, and `/bin/sh` did not exist everywhere,...